app/tools/image_gen.py

```
@@ -62,3 +73,40 @@ def generate_image(user_prompt: str) -> str:
+    if not base64_img:
+        raise ValueError("Image generation failed or image not found in response")
+    
+    try:
+        img_bytes = base64.b64decode(base64_img)
+        image_path = f"/tmp/gen_{uuid4().hex}.png"
+        with open(image_path, 'wb') as f:
+            f.write(img_bytes)
+
+        return image_path
+    except Exception as e:
+        raise RuntimeError(f"Failed to write or save image: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)